taskvine/src/bindings/python3/ndcctools/taskvine/dask_executor.py

```diff
@@ -281,7 +281,9 @@ def _dask_execute(self, dsk, keys):
                         print(f"{t.key} ran on {t.hostname}")
 
                     if t.successful():
-                        self.category_execution_time[t.category].append(t.execution_time)
+                        cat_total_time = self.category_execution_time[t.category]["avg_execution_time"] * self.category_execution_time[t.category]["num_tasks"]
+                        self.category_execution_time[t.category]["num_tasks"] += 1
+                        self.category_execution_time[t.category]["avg_execution_time"] = (cat_total_time + t.resources_measured.wall_time) / self.category_execution_time[t.category]["num_tasks"]
```

Comment:
It is easier and more numerically precise to simply keep the total execution time and compute the average only when needed.

Reply:
@btovar Thanks! That's a very good point!